src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -125,9 +125,13 @@ RTPSParticipantImpl::RTPSParticipantImpl(
     // Workaround TCP discovery issues when register
     switch (PParam.builtin.discovery_config.discoveryProtocol)
     {
+        case DiscoveryProtocol::BACKUP:
+            m_persistence_guid = m_guid;
+            // keep setting up transport
+            #pragma warning(suppress:5030)
+            [[clang::fallthrough]];
```

Comment:
As `[[fallthrough]]` is C++17, and not quite recommended by [core guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es78-dont-rely-on-implicit-fallthrough-in-switch-statements) do not recommend to rely on fallthroughs, I would suggest to create a `static void check_params_for_discovery()` method, and call it on both.

Reply:
Ok, let's remove the `[[fallthrough]]`. But I believe the changes lack entity enough (single line) to create an independent method (that would mess thinks instead of clarifying them).